src/mod2c_core/nocpout.c

```diff
@@ -1074,6 +1044,93 @@ static const char *_mechanism[] = {\n\
 		q=q->next->next->next;
 	}
 	
+#if BBCORE
+    // start printing struct that contail all global variables
+    Lappendstr(globals_update_list, "struct _global_variables_t : public MemoryManaged {\n");
+    for(int i=0; i < global_variables_count; i++) {
+        // individual member declaration: type name [len]
+        if(global_variables[i].is_array) {
+            Sprintf(buf, "  %s %s[%d];\n",
+                    global_variables[i].type,
+                    global_variables[i].name,
+                    global_variables[i].array_length);
+        } else {
+            Sprintf(buf, "  %s %s;\n",
+                    global_variables[i].type,
+                    global_variables[i].name);
+        }
+        Lappendstr(globals_update_list, buf);
+    }
+    //TODO: always have _ml_mechtype variable but it seems like we don't
+    // need to use _mechtype anywhere in the GPU kernel. So remove this.
+    Lappendstr(globals_update_list, "  int _ml_mechtype;");
+    Lappendstr(globals_update_list, "};\n\n");
```

Comment:
note to myself: remove this and see if all tests passes.

Reply:
Done in 1264bc3.